compiler/src/dmd/traits.d

```diff
@@ -1081,56 +1102,59 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
                  e.ident == Id.getVirtualMethods ||
                  e.ident == Id.getOverloads)
         {
-            uint errors = global.errors;
-            Expression eorig = ex;
-            ex = ex.expressionSemantic(scx);
-            if (errors < global.errors)
-                e.error("`%s` cannot be resolved", eorig.toChars());
-
-            if (e.ident == Id.getVirtualFunctions)
+            if (e.ident != Id.getOverloads || dim != 1)
             {
-                // @@@DEPRECATED2.121@@@
-                // Deprecated in 2.101 - Can be removed from 2.121
-                e.deprecation("`traits(getVirtualFunctions)` is deprecated. Use `traits(getVirtualMethods)` instead");
-            }
+                uint errors = global.errors;
+                Expression eorig = ex;
+                ex = ex.expressionSemantic(scx);
+                if (errors < global.errors)
+                    e.error("`%s` cannot be resolved", eorig.toChars());
 
-            /* Create tuple of functions of ex
-             */
-            auto exps = new Expressions();
-            Dsymbol f;
-            if (auto ve = ex.isVarExp)
-            {
-                if (ve.var.isFuncDeclaration() || ve.var.isOverDeclaration())
-                    f = ve.var;
-                ex = null;
-            }
-            else if (auto dve = ex.isDotVarExp)
-            {
-                if (dve.var.isFuncDeclaration() || dve.var.isOverDeclaration())
-                    f = dve.var;
-                if (dve.e1.op == EXP.dotType || dve.e1.op == EXP.this_)
+                if (e.ident == Id.getVirtualFunctions)
+                {
+                    // @@@DEPRECATED2.121@@@
+                    // Deprecated in 2.101 - Can be removed from 2.121
+                    e.deprecation(
+                        "`traits(getVirtualFunctions)` is deprecated. Use `traits(getVirtualMethods)` instead");
```

Comment:
Did this really need to be broken up into two lines?

Just viewing with whitespace off, having nothing technical to say, so just making suggestions to make the total number of changes down to a minimum.

Reply:
[D code style](https://dlang.org/dstyle.html#phobos_line_length) says hard 120 limit.

Reply:
:shrug: 
```
(master) $ wc -L compiler/src/dmd/*.d  | sort -hr
    340 compiler/src/dmd/dmangle.d
    271 compiler/src/dmd/dinterpret.d
    269 compiler/src/dmd/func.d
    266 compiler/src/dmd/dcast.d
    265 compiler/src/dmd/expressionsem.d
    257 compiler/src/dmd/toctype.d
    235 compiler/src/dmd/dtemplate.d
    230 compiler/src/dmd/dmacro.d
    223 compiler/src/dmd/ctfeexpr.d
    211 compiler/src/dmd/doc.d
    204 compiler/src/dmd/dsymbolsem.d
    193 compiler/src/dmd/parse.d
    192 compiler/src/dmd/objc_glue.d
    192 compiler/src/dmd/hdrgen.d
    190 compiler/src/dmd/errors.d
    189 compiler/src/dmd/astbase.d
    188 compiler/src/dmd/access.d
    185 compiler/src/dmd/mtype.d
    183 compiler/src/dmd/typesem.d
    183 compiler/src/dmd/todt.d
    178 compiler/src/dmd/dimport.d
    178 compiler/src/dmd/cppmanglewin.d
    177 compiler/src/dmd/libelf.d
    176 compiler/src/dmd/cparse.d
    172 compiler/src/dmd/semantic3.d
    172 compiler/src/dmd/optimize.d
    170 compiler/src/dmd/globals.d
    169 compiler/src/dmd/expression.d
    167 compiler/src/dmd/tocsym.d
    164 compiler/src/dmd/opover.d
    163 compiler/src/dmd/traits.d
    162 compiler/src/dmd/sideeffect.d
    161 compiler/src/dmd/initsem.d
    160 compiler/src/dmd/toir.d
    160 compiler/src/dmd/cli.d
    159 compiler/src/dmd/dsymbol.d
    159 compiler/src/dmd/declaration.d
    154 compiler/src/dmd/escape.d
    154 compiler/src/dmd/dtoh.d
    153 compiler/src/dmd/vsoptions.d
    153 compiler/src/dmd/e2ir.d
    151 compiler/src/dmd/statement.d
    149 compiler/src/dmd/semantic2.d
    148 compiler/src/dmd/dinifile.d
    146 compiler/src/dmd/link.d
    145 compiler/src/dmd/lexer.d
    145 compiler/src/dmd/dmodule.d
    145 compiler/src/dmd/constfold.d
    144 compiler/src/dmd/blockexit.d
    143 compiler/src/dmd/toobj.d
    143 compiler/src/dmd/target.d
    142 compiler/src/dmd/mars.d
    142 compiler/src/dmd/clone.d
    141 compiler/src/dmd/statementsem.d
    139 compiler/src/dmd/glue.d
    137 compiler/src/dmd/scanmscoff.d
    137 compiler/src/dmd/dclass.d
    137 compiler/src/dmd/attrib.d
    137 compiler/src/dmd/astenums.d
    136 compiler/src/dmd/aggregate.d
    135 compiler/src/dmd/eh.d
    135 compiler/src/dmd/dstruct.d
    133 compiler/src/dmd/dscope.d
    133 compiler/src/dmd/cppmangle.d
    132 compiler/src/dmd/libmscoff.d
    132 compiler/src/dmd/iasmdmd.d
    132 compiler/src/dmd/chkformat.d
    131 compiler/src/dmd/scanmach.d
    131 compiler/src/dmd/nogc.d
    130 compiler/src/dmd/ob.d
    130 compiler/src/dmd/json.d
    130 compiler/src/dmd/frontend.d
    129 compiler/src/dmd/statement_rewrite_walker.d
    128 compiler/src/dmd/inline.d
    126 compiler/src/dmd/cpreprocess.d
    124 compiler/src/dmd/scanomf.d
    123 compiler/src/dmd/entity.d
    122 compiler/src/dmd/libmach.d
    122 compiler/src/dmd/intrange.d
    122 compiler/src/dmd/argtypes_sysv_x64.d
    121 compiler/src/dmd/tokens.d
```

Reply:
It's not great lol, but I figured I shouldn't actively make it worse.